src/Core/Crawl/controlpos.cpp

```diff
@@ -0,0 +1,109 @@
+//#
+//# Copyright (C) 2021-2021 QuasarApp.
+//# Distributed under the GPLv3 software license, see the accompanying
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#include "controlpos.h"
+#include "clasteritem.h"
+#include <QtMath>
+
+namespace CRAWL {
+
+ControlPos::ControlPos() {
+
+}
+
+void ControlPos::add(ClasterItem *object) {
+
+    Claster::add(object);
+    updatePosition();
+
+}
+
+void ControlPos::remove(ClasterItem *object) {
+
+    Claster::remove(object->guiId());
+    updatePosition();
+
+}
+
+void ControlPos::changeLayout(const Refresh &fig) {
+    _shape = fig;
+}
+
+void ControlPos::setDistance(int dist) {
+    _distance = dist;
+}
```

Comment:
you must refresh positions of all objects after change base configuration
```suggestion
void ControlPos::changeLayout(const Refresh &fig) {
    _shape = fig;
    updatePosition();
}

void ControlPos::setDistance(int dist) {
    _distance = dist;
    updatePosition();
}
```